DemocracyClub/WhoCanIVoteFor
# Feature/party pages update

[Trello](https://trello.com/c/gziXi9a7/3366-wcivf-party-pages-part-ii)
This PR is a somewhat fiddly collection of little things. Currently blocked by [yournextrepresentative#2100](https://github.com/DemocracyClub/yournextrepresentative/pull/2100). 

### To test
#### Pre-test - YNR

_There's an amount of setup to do because the YNR changes haven't been merged yet - sorry :(_ 

- Check out `feat/update-party-importer`
- Run the migrations 
- Run `python manage.py parties_import_from_ec`
- Run `runserver 4444` - this must stay running for the duration of your local testing!

#### Pre-test - WCIVF
- Check out this branch (`feature/party-pages-update`)
- In base.py, change `YNR_BASE` to `http://0.0.0.0:4444`
- Run the migrations
- Run `python manage.py import_parties`
- `runserver`

#### Test 1: Footer
- Criteria: A link in the footer exists which takes you to /parties/
- Open any page 
- Scroll down to the footer
- There should be a link for `All Parties`
- Click the link
- Link takes you to `http://localhost:8000/parties/`

<img width="950" alt="Screenshot 2023-06-16 at 15 11 24" src="https://github.com/DemocracyClub/WhoCanIVoteFor/assets/9531063/217d0944-0991-4e9e-a58c-4f038e4686da">


#### Test 2: Alternative name
- Criteria: Where a party has an alternative name, it is shown near the top of the page and is included as the first ballot description. 
- Suggested party: http://localhost:8000/parties/party:53/labour-party
- Go to the page of a party with an alternative name 
- You should see something like this, with the alternative name under the main header
<img width="564" alt="Screenshot 2023-06-16 at 15 10 20" src="https://github.com/DemocracyClub/WhoCanIVoteFor/assets/9531063/1cb42c87-c0a8-4ea6-9eb5-b8b973311019">

- Scroll down to `Ballot Descriptions`
- The top bullet point in this section should be `{party name} | {alternative_name}` as in this screenshot
<img width="833" alt="Screenshot 2023-06-16 at 15 27 37" src="https://github.com/DemocracyClub/WhoCanIVoteFor/assets/9531063/8ae66e76-76dc-446f-ae7d-e475ff603628">

#### Test 3: Descriptions
- Criteria: Parties with no active descriptions and no registered alternative name should show "This party has no current alternative descriptions registered with the Electoral Commission.", parties with inactive descriptions should have a "Previous Descriptions" detail view
- Suggested party: http://localhost:8000/parties/party:2302/
- Go to the page of a party with inactive descriptions
- Scroll down to `Ballot Descriptions`
- You should see a detail view, titled `Previous Descriptions`, as in the screenshot below
<img width="962" alt="Screenshot 2023-06-16 at 15 54 01" src="https://github.com/DemocracyClub/WhoCanIVoteFor/assets/9531063/95d433ff-aec4-46de-b553-97cc6943f802">


- Find a party who has no alternative name and no active descriptions
- You should see "This party has no current alternative descriptions registered with the Electoral Commission.", as in the screenshot above.

#### Test 4: Emblems
- Criteria: Parties with no active emblems should show "This party has no current alternative emblems registered with the Electoral Commission.", parties with inactive emblems should have a "Previous Emblems" detail view
- Suggested party: http://localhost:8000/parties/party:7947/
- Go to the page of a party with inactive emblems 
- Scroll down to `Emblems`
- You should see a detail view, titled `Previous Emblems` as in the screenshot below
<img width="969" alt="Screenshot 2023-06-16 at 15 54 10" src="https://github.com/DemocracyClub/WhoCanIVoteFor/assets/9531063/04514b2f-c206-4eb9-b4a7-284ae7b95745">

- Find a party with no active emblems
- You should see "This party has no current alternative emblems registered with the Electoral Commission.", as in the screenshot above

```[tasklist]
### PR Checklist



Check off items once the PR addresses them.

```
